pkg/provider/aws/provider.go

```diff
@@ -60,8 +61,11 @@ func NewProvider() (provider.Provider, error) {
 	}, nil
 }
 
-func retrieveInstanceNetworkInterfaceMacAddress() (string, error) {
-	res, err := http.Get(instanceMetadataEndpoint + "/mac")
+func retrieveInstanceNetworkInterfaceMacAddress(token string) (string, error) {
+	client := &http.Client{}
```

Comment:
The client should probably be a value and have a timeout.
```suggestion
	client := http.Client{Timeout: 3*time.Second}
```
Also, it would be nice to reuse the client in the following calls instead of recreating one.

Reply:
Why a value? Why a timeout? If you don't put a timeout, I'm pretty sure it'll have one anyway, much bigger though.

Reply:
No, the default http.Client does not have any timeout https://medium.com/@nate510/don-t-use-go-s-default-http-client-4804cb19f779

Reply:
fixed